### ISSUE TITLE
Add mining-pool client support w/ fallback to zcashd.

### ISSUE THREAD

**Issue by gesker** [zcash/zcash#1913]
First, thank you to the ZCash team for this project. I feel that it is a solid step forward for crypto-currency.

However, using the most recent version in the deb repository (1.0.3) I haven't been able to use flypool with the -stratum option. I ended up falling back to 3rd party nheqminer.

I would like to be able to:

1. Run zcashd in daemon mode on multiple machines (already doable of course)
2. Simultaneously have each of these daemons participate in mining pools like flypool.org -- or even better yet  -- one run by the zcash project itself.
3. If for some reason the daemon cannot reach the mining pool have the daemon fall back into solo mining but still designate the specific wallet for mining results
4. Attempt to reconnect/recover to designated pool at regular intervals when in solo mining mode

In short it would be great if the zcashd was a _one stop shop_ for node, wallet and ALL mining activity -- cpu, gpu if available, pool, solo, recovery from solo to pool, etc.

Again, thanks for this project!

Cordially,
Dennis

**Comment by sammy007** [comment]
In-daemon mining is a nonsense, only suitable for several hours after launch. If you still need pool mining with failover to solo, use external miners with support for failovers or haproxy with one upstream configured to a pool of your choice and another upstream to your own private stratum setup.

**Comment by gesker** [comment]
Hi sammy007:

Why is in-daemon mining nonsense? Why is the core/official daemon of the poject having the capability to participate in a mining pool a bad thing? There is already a -stratum option in zcashd?

Could you expand on this a bit so I can better understand your perspective? I'm just an end user of zcashd and do not claim expertise in blockchain or the inner workings of this crypto currency iteration.

Are you a member of the zcash project? Is your position on in-daemon mining and the use of pool miners the official position of the zcash project?

Thanks,
Dennis

**Comment by sammy007** [comment]
It will take 1 year to find a block with single GPU. CPU mining is dead since recent GPU optimizations. So in-daemon mining is for fun (very questionable fun tbh).

Of course I am not a member of project.

**Comment by ioptio** [comment]
Greetings, thanks for the feedback! I do like the idea of defaulting to solo mining when pool connection fails... regardless of profitability of solo mining, keeping nodes up and running while trying to reconnect to the pool would benefit network security. Regarding zcashd as a "one stop shop", I can see where you're coming from but we're weary of having folks depend on ZcashCo for all Zcash related software. We're extremely grateful for the ecosystem to build alternative clients so users have choice. Similar reasons for why we probably won't ever run our own pool.

**Comment by sammy007** [comment]
You can use my https://github.com/sammy007/zcash-proxy for solo, it's extremely easy to install on Linux. It has 2.5% devfee (block reward split) and it's closed source, but it's easiest solution in crypto space and great candidate for use cases like backup solo pool.

**Comment by nathan-at-least** [comment]
@str4d mentioned how his standalone miner fork may be useful for this request, so I assigned him. ;-)

**Comment by nuttycom** [comment]
Pool support has been provided by 3rd-party mining pool software for many years, and falling back to zcashd's CPU mining is not a useful alternative given the dominance of ASIC miners on the network.
